web/crux/src/domain/agent.spec.ts

```diff
@@ -0,0 +1,380 @@
+
+      await expect(deleteChannel).resolves.not.toThrow()
+    })
+
+    it('delete container command should timeout if without response', async () => {
+      const commandChannel = firstValueFrom(agent.onConnected())
+
+      const deleteRequest: DeleteContainersRequest = {
+        prefix: 'prefix',
+      }
+
+      const deleteChannel = firstValueFrom(agent.deleteContainers(deleteRequest))
+
+      await expect(commandChannel).resolves.toEqual({
+        deleteContainers: deleteRequest,
+      })
+
+      await expect(deleteChannel).resolves.toEqual(Empty)
+    })
+  })
+
+  describe('update', () => {
+    it('should send update command', async () => {
+      const commandChannel = firstValueFrom(agent.onConnected())
+
+      await expect(agent.updating).toEqual(false)
+
+      agent.update('update-tag')
+
+      await expect(agent.updating).toEqual(true)
+
+      await expect(commandChannel).resolves.toEqual({
+        update: {
+          tag: 'update-tag',
+          timeoutSeconds: 60 * 5 * 1000,
+        },
+      })
+    })
+
+    it('should throw exception while updating', () => {
+      agent.update('update-tag')
+
+      expect(() => agent.update('update-tag')).toThrow(CruxPreconditionFailedException)
+    })
+
+    it('abort update should return agent to connected state', async () => {
+      const eventChannelPromise = firstValueFrom(eventChannel.pipe(skip(1)))
+
+      agent.onConnected()
+
+      agent.update('update-tag')
+
+      agent.onUpdateAborted('test')
+
+      await expect(agent.updating).toEqual(false)
+
+      await expect(eventChannelPromise).resolves.toEqual({
+        id: AGENT_ID,
+        status: 'connected',
+        error: 'test',
+        updating: false,
+      })
+    })
+  })
+
+  describe('secrets', () => {
+    it('getting container secrets should send command to agent and resolve with secrets', async () => {
+      const commandChannel = firstValueFrom(agent.onConnected())
+
+      const secrets = firstValueFrom(agent.getContainerSecrets('prefix', 'name'))
+
+      await expect(commandChannel).resolves.toEqual({
+        listSecrets: {
+          prefix: 'prefix',
+          name: 'name',
+        },
+      })
+
+      const message = {
+        prefix: 'prefix',
+        name: 'name',
+        publicKey: 'key',
+        hasKeys: true,
+        keys: ['k1', 'k2', 'k3'],
+      }
+
+      agent.onContainerSecrets(message)
+
+      await expect(secrets).resolves.toEqual(message)
+    })
+
+    it('getting container secrets should send command to agent and timeout without a response', async () => {
+      const commandChannel = firstValueFrom(agent.onConnected())
+
+      const secrets = firstValueFrom(agent.getContainerSecrets('prefix', 'name'))
+
+      await expect(commandChannel).resolves.toEqual({
+        listSecrets: {
+          prefix: 'prefix',
+          name: 'name',
+        },
+      })
+
+      await expect(secrets).rejects.toThrow()
+    }, 10000)
```

Comment:
please put a comment here why 10000
export the agent's constant as we discussed in person